lib/practice_exercises.rb

```diff
@@ -1,13 +1,48 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) as the time taken will depend on the length of list.
+# Space Complexity: I'd like to say O(1) as the space the array takes does not grow. 
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  if list.length == 0 || list.length == 1
+    return list 
+  end 
+
+  i = 0
+  until i == list.length
+    if list[i] == list[i+1]
+      list.delete_at(i)
+    end 
+    i += 1
+  end 
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) since the time it takes to process depends on the length of the string of arrays 
```

Comment:
It also depends on the length of the strings, so I would say O(n * m).  If you assume the strings to be small, then you can say O(n), but you should note that assumption.